src/main/services/additional-content/maps/local-maps-manager.service.ts

```diff
@@ -306,6 +309,85 @@ export class LocalMapsManagerService {
         return null;
     }
 
+    public importMaps(zipPaths: string[], version?: BSVersion): Observable<Progression> {
+        const progress: Progression<string> = {
```

Comment:
This function should handle zips containing multiple maps, not just one zip per map.\
It should be compatible with the export maps feature of BSM, which creates a single zip file for all the selected maps.

Reply:
Is there a way to currently verify that the exportMaps zip file is generated by BSM? Something like a `metadata` within the zip file.

Also the only problem that I can see here is the **zip bomb check** I've implemented in processZip. Where there is a limit to the file count and file size. We can add that information in within the metadata I guess to bypass the check and a warn users about downloading other users exportMaps zip files.

Reply:
We can add a metadata file to the generated zip, but users can totally create zips that contain multiple maps on their own for plenty of reasons.

We can maybe split the zip verification to another ipc, and warn the user with a modal if the zip exeed a certain size or a certain amount of maps 🤔 

Or we could let the user be responsible for what kind of zips they are importing (I mean, if they want to import 1TB of maps, they should be able to do it)

Reply:
> We can add a metadata file to the generated zip, but users can totally create zips that contain multiple maps on their own for plenty of reasons.

Yeah fair point, the metadata would be either useless or redundant in any case. Might just do a simple verification on each folder on depth 1 if they contain the "I/info.dat" file.

> We can maybe split the zip verification to another ipc, and warn the user with a modal if the zip exeed a certain size or a certain amount of maps 🤔
Or we could let the user be responsible for what kind of zips they are importing (I mean, if they want to import 1TB of maps, they should be able to do it)

I think the latter is probably the better option since the user should be responsible for what the zip files that are being imported is.

I've implemented that check is because sonarcloud suggested it so might ignore that in the future in any case.

Reply:
Yeah, I think SonarCloud's suggestion is great when it’s for a website, but it doesn’t really apply to BSM